index.go

```diff
@@ -0,0 +1,275 @@
+	PrimaryKey string
+
+	client *Client
+}
+
+type IndexInterface interface {
+	FetchInfo() (resp *Index, err error)
+	FetchPrimaryKey() (resp *string, err error)
+	UpdateIndex(primaryKey string) (resp *Index, err error)
+	Delete(uid string) (ok bool, err error)
+	GetStats() (resp *StatsIndex, err error)
+
+	AddDocuments(documentsPtr interface{}) (resp *AsyncUpdateID, err error)
+	UpdateDocuments(documentsPtr interface{}) (*AsyncUpdateID, error)
+	AddDocumentsWithPrimaryKey(documentsPtr interface{}, primaryKey string) (resp *AsyncUpdateID, err error)
+	UpdateDocumentsWithPrimaryKey(documentsPtr interface{}, primaryKey string) (*AsyncUpdateID, error)
+	GetDocument(identifier string, documentPtr interface{}) error
+	GetDocuments(request DocumentsRequest, response interface{}) error
+	DeleteDocument(identifier string) (resp *AsyncUpdateID, err error)
+	DeleteDocuments(identifier []string) (resp *AsyncUpdateID, err error)
+	DeleteAllDocuments() (resp *AsyncUpdateID, err error)
+	Search(request SearchRequest) (*SearchResponse, error)
+
+	GetUpdateStatus(updateID int64) (resp *Update, err error)
+	GetAllUpdateStatus() (resp []Update, err error)
+
+	GetSettings() (resp *Settings, err error)
+	UpdateSettings(request Settings) (resp *AsyncUpdateID, err error)
+	ResetSettings() (resp *AsyncUpdateID, err error)
+	GetRankingRules() (resp *[]string, err error)
+	UpdateRankingRules(request []string) (resp *AsyncUpdateID, err error)
+	ResetRankingRules() (resp *AsyncUpdateID, err error)
+	GetDistinctAttribute() (resp *string, err error)
+	UpdateDistinctAttribute(request string) (resp *AsyncUpdateID, err error)
+	ResetDistinctAttribute() (resp *AsyncUpdateID, err error)
+	GetSearchableAttributes() (resp *[]string, err error)
+	UpdateSearchableAttributes(request []string) (resp *AsyncUpdateID, err error)
+	ResetSearchableAttributes() (resp *AsyncUpdateID, err error)
+	GetDisplayedAttributes() (resp *[]string, err error)
+	UpdateDisplayedAttributes(request []string) (resp *AsyncUpdateID, err error)
+	ResetDisplayedAttributes() (resp *AsyncUpdateID, err error)
+	GetStopWords() (resp *[]string, err error)
+	UpdateStopWords(request []string) (resp *AsyncUpdateID, err error)
+	ResetStopWords() (resp *AsyncUpdateID, err error)
+	GetSynonyms() (resp *map[string][]string, err error)
+	UpdateSynonyms(request map[string][]string) (resp *AsyncUpdateID, err error)
+	ResetSynonyms() (resp *AsyncUpdateID, err error)
+	GetAttributesForFaceting() (resp *[]string, err error)
+	UpdateAttributesForFaceting(request []string) (resp *AsyncUpdateID, err error)
+	ResetAttributesForFaceting() (resp *AsyncUpdateID, err error)
+
+	WaitForPendingUpdate(ctx context.Context, interval time.Duration, updateID *AsyncUpdateID) (UpdateStatus, error)
+	DefaultWaitForPendingUpdate(updateID *AsyncUpdateID) (UpdateStatus, error)
+}
+
+var _ IndexInterface = &Index{}
+
+func newIndex(client *Client, uid string) *Index {
+	i := &Index{
+		UID:    uid,
+		client: client,
+	}
+	return i
```

Comment:
Unnecesary assignation
```suggestion
	return &Index{
		UID:    uid,
		client: client,
	}
```